nodebuilder/tests/blob_test.go

```diff
@@ -123,6 +123,37 @@ func TestBlobModule(t *testing.T) {
 				require.ErrorContains(t, err, blob.ErrBlobNotFound.Error())
 			},
 		},
+		{
+			name: "Submit equal blobs",
+			doFn: func(t *testing.T) {
+				appBlob, err := blobtest.GenerateV0Blobs([]int{8, 4}, true)
+				require.NoError(t, err)
+				b, err := blob.NewBlob(
+					appBlob[0].ShareVersion,
+					append([]byte{appBlob[0].NamespaceVersion}, appBlob[0].NamespaceID...),
+					appBlob[0].Data,
+				)
+				require.NoError(t, err)
+				height, err := fullClient.Blob.Submit(ctx, []*blob.Blob{b, b}, nil)
+				require.NoError(t, err)
+				_, err = fullClient.Header.WaitForHeight(ctx, height)
+				require.NoError(t, err)
+				b0, err := fullClient.Blob.Get(ctx, height, b.Namespace(), b.Commitment)
+				require.NoError(t, err)
+				require.Equal(t, b, b0)
```

Comment:
Irrc, the Equal just compares the pointers, while EqualValues would do deep comparison. 

I might be wrong, but could you verify b and b0 are actually different struct in men?

Also, let's add a case where we see that the namespace does have two equal blobs. It might deduplicate the same blobs on in a single PFB, but definitely doesn't deduplicate blobs if they are sent in different PFBs. In the test case we send both blobs in a single PFB

Reply:
```
// EqualValues asserts that two objects are equal or convertable to the same types
// and equal.
```
You are right, it checks the raw types

Reply:
Yes and I am concerned that simple Equals succeeds here

Reply:
Double-checked: `Equal` also checks the underlying values in case of pointers:

```
// Equal asserts that two objects are equal.
// Pointer variable equality is determined based on the equality of the
// referenced values (as opposed to the memory addresses)
```

Reply:
I see, thanks.

Please also address this comment 
> Also, let's add a case where we see that the namespace does have two equal blobs. It might deduplicate the same blobs on in a single PFB, but definitely doesn't deduplicate blobs if they are sent in different PFBs. In the test case we send both blobs in a single PFB

Reply:
I'm a bit confused: do we expect an error on the second submit call?

Reply:
added one more case